standard/logic.lua

```diff
@@ -119,6 +121,45 @@ function Logic.try(f)
 	return require('Module:ResultOrError').try(f)
 end
 
+---Returns the result of a function if successful. Otherwise it returns the result of the second function.
+---If the first function fails, its error is logged to the console and stashed away for display.
+---@param f fun(): any
+---@param other? fun(error: error): any
+---@param makeError? fun(error: error): error function that allows customizing Error instance being logged and stashed.
+---@return any?
+function Logic.tryOrElseLog(f, other, makeError)
+	return Logic.try(f)
+		:catch(function(error)
+			if type(error) == 'string' then
+				error = Error(error)
+			end
+
+			error.header = 'Error occured while calling a function: (caught by Logic.tryOrElseLog)'
+			if makeError then
+				error = makeError(error)
+			end
+
+			require('Module:Error/Ext').logAndStash(error)
+
+			if other then
+				return other(error)
+			end
+		end)
+		:get()
+end
+
+---Returns the result of a function if successful. Otherwise it returns nil.
+---If the first function fails, its error is logged to the console and stashed away for display.
+---@param f fun(): any
+---@param makeError? fun(error: error): error function that allows customizing Error instance being logged and stashed.
+---@return function
+function Logic.wrapTryOrLog(f, makeError)
+	return function(...)
+		local args = require('Module:Table').pack(...)
+		return Logic.tryOrElseLog(function() return f(unpack(args)) end, nil, makeError)
```

Comment:
looks like it is so that we can actually safe it^^
![Screenshot 2023-07-24 14 41 13](https://github.com/Liquipedia/Lua-Modules/assets/75081997/a7e13eec-2a62-49e7-88f7-16f2c15b5c13)